### PR TITLE
WIP chore(modules): bump k8s to 1.22-beta.2

### PR DESCRIPTION
**Pull request by njhale**
Bump k8s dependencies to `1.22-beta.2` and remove remaining references to APIs set to be removed in `1.22.0`.

Part of [OLM-2222][1]

[1]: https://issues.redhat.com/browse/OLM-2222

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,10 +11,10 @@ require (
...
 	github.com/spf13/cobra v1.1.3
 	github.com/stretchr/testify v1.7.0
-	k8s.io/api v0.21.1
-	k8s.io/apiextensions-apiserver v0.21.1
-	k8s.io/apimachinery v0.21.1
-	k8s.io/client-go v0.21.1
+	k8s.io/api v0.22.0-beta.2
+	k8s.io/apiextensions-apiserver v0.22.0-beta.2
+	k8s.io/apimachinery v0.22.0-beta.2
+	k8s.io/client-go v0.22.0-beta.2
```
Comment (timflannagan) [inline go.mod:17]: IIRC, we need to also re-generate the CRD manifests as bumping these dependencies have updated (IIRC description fields?) fields. It looks like this is why the verify check is failing. Likely need to run `make manifests` and/or `make generate` locally and commit the results.

**Comment by timflannagan** [comment]
I missed there's a decent amount of v1beta1 CRD references in this repository.

/lgtm cancel

**Comment by estroz** [comment]
Can bump to 1.22 now.

**Pull request closed without merging**